File: homework/homework.cpp
#include <iostream>
#include <fstream>
#include <algorithm>
using namespace std;

int q[100000];
int main(void) {
	ifstream input;
	input.open("homework.in");
	int N, sum;
	input>>N;
	for (int i = 0; i<N; i++) {
		input>>q[i];
		sum+=q[i];
	}
	input.close();

	int max = 0, partialSum = 0, min = 0;
	double avg = 0;
	for (int i = 1; i<N-2; i++) {
		partialSum+=q[i];
		min = *min_element(q+i, q+N);
		avg = ((double)(sum-partialSum-min))/(N-i-1);
		if (avg>max) {
			max = avg;
		}
	}

	ofstream output;
	output.open("homework.out");
	partialSum = 0; min = 0; avg = 0;
	for (int i = 1; i<N-2; i++) {
		partialSum+=q[i];
		min = *min_element(q+i, q+N);
		avg = ((double)(sum-partialSum-min))/(N-i-1);
		if (avg==max) {
			output<<i<<endl;
		}
	}
	output.close();
	return 0;
}

File: homework/homework.py
fin = open('homework.in')
data = fin.readlines()
fin.close()
N = int(data[0][:-1])
q = [int(i) for i in data[1].split(' ')]

s = sum(q)
partial=0
avgs=[0]*(N-1)
for i in range(1, N-1):
	partial+=q[i-1]
	avgs[i-1] = (s-partial-min(q[i:]))/(N-i-1)
print(avgs)

#def computeAvg(K):
#	return sum([i for i in q[K:] if i != min(q[K:])])/(N-K-1)

#avgs = [computeAvg(i) for i in range(1, N-1)]
maxAvg = max(avgs)
fout = open('homework.out', 'w')
for i in range(1, N-1):
	if avgs[i-1]==maxAvg:
		fout.write(str(i))
fout.close()
